lib/types/api.types.ts

```diff
@@ -271,56 +292,70 @@ export type DialogAppSDK = Omit<BaseAppSDK, 'ids'> & {
+> = Omit<BaseAppSDK<InstallationParameters, InstanceParameters, never>, 'ids'> & {
   ids: Omit<IdsAPI, EntryScopedIds>
 }
 
-export type ConfigAppSDK = Omit<BaseAppSDK, 'ids'> & {
+export type ConfigAppSDK<
+  InstallationParameters extends KeyValueMap = KeyValueMap,
+  InstanceParameters extends KeyValueMap = KeyValueMap
+> = Omit<BaseAppSDK<InstallationParameters, InstanceParameters, never>, 'ids'> & {
   /** A set of IDs actual for the app */
   ids: Omit<IdsAPI, EntryScopedIds | 'extension' | 'app'> & { app: string }
   app: AppConfigAPI
 }
 
-export type KnownAppSDK =
-  | FieldAppSDK
-  | SidebarAppSDK
-  | DialogAppSDK
-  | EditorAppSDK
-  | PageAppSDK
-  | ConfigAppSDK
-  | HomeAppSDK
+export type KnownAppSDK<
+  InstallationParameters extends KeyValueMap = KeyValueMap,
+  InstanceParameters extends KeyValueMap = KeyValueMap,
+  InvocationParameters extends SerializedJSONValue = SerializedJSONValue
+> =
+  | FieldAppSDK<InstallationParameters, InstanceParameters>
+  | SidebarAppSDK<InstallationParameters, InstanceParameters>
+  | DialogAppSDK<InstallationParameters, InstanceParameters, InvocationParameters>
+  | EditorAppSDK<InstallationParameters, InstanceParameters>
+  | PageAppSDK<InstallationParameters, InstanceParameters, InvocationParameters>
+  | ConfigAppSDK<InstallationParameters, InstanceParameters>
+  | HomeAppSDK<InstallationParameters, InstanceParameters>
 
 /** @deprecated consider using {@link BaseAppSDK} */
-export type BaseExtensionSDK = BaseAppSDK
+export type BaseExtensionSDK = BaseAppSDK<KeyValueMap, KeyValueMap, never>
 
 /** @deprecated consider using {@link EditorAppSDK} */
-export type EditorExtensionSDK = EditorAppSDK
+export type EditorExtensionSDK = EditorAppSDK<KeyValueMap, KeyValueMap>
 
 /** @deprecated consider using {@link SidebarAppSDK} */
-export type SidebarExtensionSDK = SidebarAppSDK
+export type SidebarExtensionSDK = SidebarAppSDK<KeyValueMap, KeyValueMap>
 
 /** @deprecated consider using {@link FieldAppSDK} */
-export type FieldExtensionSDK = FieldAppSDK
+export type FieldExtensionSDK = FieldAppSDK<KeyValueMap, KeyValueMap>
 
 /** @deprecated consider using {@link DialogAppSDK} */
-export type DialogExtensionSDK = DialogAppSDK
+export type DialogExtensionSDK = DialogAppSDK<KeyValueMap, KeyValueMap>
 
 /** @deprecated consider using {@link PageAppSDK} */
-export type PageExtensionSDK = PageAppSDK
+export type PageExtensionSDK = PageAppSDK<KeyValueMap, KeyValueMap, { path: string }>
 
 /** @deprecated consider using {@link HomeAppSDK} */
-export type HomeExtensionSDK = HomeAppSDK
+export type HomeExtensionSDK = HomeAppSDK<KeyValueMap, KeyValueMap>
 
 /** @deprecated consider using {@link ConfigAppSDK} */
-export type AppExtensionSDK = ConfigAppSDK
+export type AppExtensionSDK = ConfigAppSDK<KeyValueMap, KeyValueMap>
 
 /** @deprecated consider using {@link KnownAppSDK} */
-export type KnownSDK = KnownAppSDK
+export type KnownSDK = KnownAppSDK<KeyValueMap, KeyValueMap, SerializedJSONValue>
```

Comment:
We don't need to specify the generic types here because the XXAppSDK types have defaults for their generics